strawberry/type.py

```diff
@@ -118,6 +127,66 @@ class StrawberryOptional(StrawberryContainer):
     ...
 
 
+class StrawberryAnnotated(StrawberryContainer):
+    """
+    Equivalent to typing.Annotated for strawberry types
+    """
+
+    def __init__(self, of_type: Union[StrawberryType, type], *args):
+        of_type, base_args = StrawberryAnnotated.get_type_and_args(of_type)
+        super().__init__(of_type)
+        self.args = base_args + list(args)
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, StrawberryType):
+            if type(self) == type(other):
+                other = cast(StrawberryAnnotated, other)
+                return (self.of_type, self.args) == (other.of_type, other.args)
+            else:
+                return False
+
+        return super().__eq__(other)
+
+    def __hash__(self) -> int:
+        return hash((type(self), self.of_type, self.args))
+
+    def __str__(self) -> str:
+        args = ", ".join(map(str, [self.of_type] + self.args))
+        return f"{type(self).__name__}({args})"
+
+    def __repr__(self) -> str:
+        args = ", ".join(map(repr, [self.of_type] + self.args))
+        return f"{type(self).__name__}({args})"
```

Comment:
Why define both `__str__` and `__repr__` identically? If `__repr__` is defined, but `__str__` is not, `repr()` will use `__str__`. 

(https://stackoverflow.com/a/2626364)